src/transform/src/extractor.ts

```diff
@@ -16,23 +16,48 @@ import {
   TypeInfo,
   typeMap,
 } from "./types.js";
+import HypermodeTransform from "./index.js";
 
 export class Extractor {
   binaryen: typeof binaryen;
   module: binaryen.Module;
   program: Program;
+  transform: Transform;
 
   constructor(transform: Transform, module: binaryen.Module) {
     this.program = transform.program;
     this.binaryen = transform.binaryen;
     this.module = module;
+    this.transform = transform;
   }
 
   getProgramInfo(): ProgramInfo {
+    if (!(this.transform instanceof HypermodeTransform)) {
```

Comment:
We don't need to support other transforms, so the scenario being checked isn't likely.